src/components/shared/dynamicFee/index.js

```diff
@@ -0,0 +1,162 @@
+import {
+  formatAmountBasedOnLocale,
+} from '../../../utils/formattedNumber';
+import { fromRawLsk } from '../../../utils/lsk';
+
+const CUSTOM_FEE_INDEX = 3;
+
+// eslint-disable-next-line max-statements
+const DynamicFee = ({
+  t,
+  token,
+  priorities,
+  selectedPriority,
+  setSelectedPriority,
+  fee,
+  customFee,
+  setCustomFee,
+}) => {
+  const [showEditIcon, setShowEditIcon] = useState(false);
+  const isCustom = selectedPriority === CUSTOM_FEE_INDEX;
+
+  const onClickPriority = (e) => {
+    e.preventDefault();
+    if (setCustomFee) {
+      setCustomFee(undefined);
+    }
+    const selectedIndex = Number(e.target.value);
+    setSelectedPriority({ item: priorities[selectedIndex], index: selectedIndex });
+    if (showEditIcon) {
+      setShowEditIcon(false);
+    }
+  };
+
+  const onInputChange = (e) => {
+    e.preventDefault();
+    const customFeeValue = e.target.value;
+    setCustomFee(customFeeValue);
+  };
+
+  const onInputBlur = (e) => {
+    e.preventDefault();
+    setCustomFee(e.target.value);
+    setShowEditIcon(true);
+  };
+
+  const onClickCustomEdit = (e) => {
+    e.preventDefault();
+    setShowEditIcon(false);
+  };
+
+  const tokenRelevantPriorities = useMemo(() =>
+    priorities.filter((_, index) =>
+      index !== CUSTOM_FEE_INDEX
+      || (index === CUSTOM_FEE_INDEX && token === tokenMap.LSK.key)),
+  [token, priorities]);
+
+  const isLoading = priorities[0].value === 0;
+
+  const getFeeStatus = () => {
+    if (customFee) {
+      return customFee;
+    }
+    return !fee.error
+      ? `${formatAmountBasedOnLocale({ value: fee.value })} ${token}`
+      : fee.feedback;
+  };
+
+  const inputValue = !isLoading && (customFee === 'undefined' ? fee.value : customFee);
+
+  return (
+    <div className={`${styles.dynamicFeeWrapper} ${styles.fieldGroup} processing-speed`}>
+      <div className={`${styles.col}`}>
+        <span className={`${styles.fieldLabel}`}>
+          {t('Priority')}
+          <Tooltip>
+            <p className={styles.tooltipText}>
+              {
+                t('Lorem Ipsum...')
+              }
+            </p>
+          </Tooltip>
+        </span>
+        <div className={`${styles.prioritySelector} priority-selector`}>
+          {tokenRelevantPriorities.map((priority, index) => (
+            <button
+              key={`fee-priority-${index}`}
+              className={`${styles.feePriority} ${index === selectedPriority ? styles.feePrioritySelected : ''} option-${priority.title}`}
+              onClick={onClickPriority}
+              value={index}
+            >
+              {priority.title}
+            </button>
+          ))}
+        </div>
+      </div>
+      <div className={`${styles.col} fee-container`}>
+        <span className={`${styles.fieldLabel}`}>
+          {t('Transaction fee')}
+          <Tooltip>
+            <p className={styles.tooltipText}>
+              {
+                t('Lorem Ipsum...')
```

Comment:
@reyraa We need tooltip text here as well.

Reply:
You can choose a high, medium, or low transaction priority, each requiring a corresponding transaction fee. Or you can pay any desired fee of no less than [minFee] [LSK/BTC]. If you don't why, we recommend you choose one of the said options instead.